### ISSUE TITLE
Structured test of the frontend

### ISSUE THREAD

**Issue by inbaz** [inbaz/ers#707]
Before we can go online we need to do some structured tests with pre defined error and exit behaviour.

- [x] Every product needs to be orderable. Therefore it needs to be able to be put into the shopping cart, edit and deleted without side effects.
- [x] It needs to be possible to place an order with every payment type.
- [x] All links in emails need to be correct and functioning.
- [x] An email address that was used for a former completed order may not be used for a second one without login. If we do that everybody can change everybodys user details. Login will not be possible on GoLive, therefore participants still need to use a second e-mail for a second order.

It would be nice to have these tests documented in a well structured document to be able to automate them later with a testing framework.

**Comment by jolieno** [comment]
yes, good idea! I'll start doing this!

**Comment by jolieno** [comment]
**Every product needs to be orderable. Therefore it needs to be able to be put into the shopping cart, edit and deleted without side effects.**  

The only thing i've found while testing this point is:

- When you enter the birthdate as a 'under age person' and later (after adding several products) edit this to a adult birthdate, the prices of product won't adjust.  I don't know if people will try to get in cheaper this way, but it would be nice to avoid this if possible.

The rest seem to work! :)

**Comment by jolieno** [comment]
**It needs to be possible to place an order with every payment type.**

I've found this so far:
- the price the system charge is even higher than yesterday, now at 52,32 €
- and the following error occurs:

You will be redirected to the credit card payment page.

        ?trx_amount=17982&trx_paymenttyp=cc&trx_redirect_url=https%3A%2F%2Fejc2016.ers.inbaz.org%2Forder%2Fthankyou%2FGY2945VPR9FXRC7UWGWNK6N8M8DKVD40&hidden_trigger_url=https%3A%2F%2Fejc2016.ers.inbaz.org%2Fpayment%2Fcc-check%2FGY2945VPR9FXRC7UWGWNK6N8M8DKVD40&trx_securityhash=0c3508c6e86bfc75fabd0d0b54eef7a1    

If the redirect is not working, please follow this link.

After this error, the filled in personal details and items in the shoppingbasket have disappeared.

**Comment by jolieno** [comment]
**All links in emails need to be correct and functioning.**

No errors occured will testing this, several time. So this point is tested and approved.
